iroh-net/src/magic_endpoint.rs

```diff
@@ -300,6 +321,53 @@ impl MagicEndpoint {
         // the packet if grease_quic_bit is set to false.
         endpoint_config.grease_quic_bit(false);
 
+        let mdns_service = if use_mdns {
+            let mdns = ServiceDaemon::new().context("mdns")?;
+            let node_id = secret_key.public();
+            // Hash the node id, to avoid publicly sharing it.
+            let service_id = hash::Hash::new(node_id);
+            let service_hostname = "iroh-discovery.local.";
+            let instance_name = format!("{}-iroh", service_id);
+            let (local_ipv4_addr, local_ipv6_addr) = msock.local_addr()?;
+            let port = local_ipv4_addr.port();
+
+            let my_addrs: Vec<_> = msock
```

Comment:
What if this info changes? Maybe I missed it, but where is the part that changes this on endpoint change?

Reply:
there is no updating happening right now, I should add at least a todo for it, yes